compiler/rustc_middle/src/ty/context/tls.rs

```diff
@@ -52,59 +47,66 @@ impl<'a, 'tcx> ImplicitCtxt<'a, 'tcx> {
     }
 }
 
-/// Sets Rayon's thread-local variable, which is preserved for Rayon jobs
-/// to `value` during the call to `f`. It is restored to its previous value after.
-/// This is used to set the pointer to the new `ImplicitCtxt`.
-#[cfg(parallel_compiler)]
-#[inline]
-fn set_tlv<F: FnOnce() -> R, R>(value: usize, f: F) -> R {
-    rayon_core::tlv::with(value, f)
-}
+mod tlv {
+    #[cfg(not(parallel_compiler))]
+    use std::cell::Cell;
 
-#[cfg(parallel_compiler)]
-#[inline]
-fn set_tlv_raw(value: usize) {
-    rayon_core::tlv::set(value);
-}
+    #[cfg(parallel_compiler)]
+    use rustc_rayon_core as rayon_core;
 
-/// Gets Rayon's thread-local variable, which is preserved for Rayon jobs.
-/// This is used to get the pointer to the current `ImplicitCtxt`.
-#[cfg(parallel_compiler)]
-#[inline]
-pub fn get_tlv() -> usize {
-    rayon_core::tlv::get()
-}
+    /// Gets Rayon's thread-local variable, which is preserved for Rayon jobs.
+    /// This is used to get the pointer to the current `ImplicitCtxt`.
+    #[cfg(parallel_compiler)]
+    #[inline]
+    pub(super) fn get_tlv() -> usize {
+        rayon_core::tlv::get()
+    }
 
-#[cfg(not(parallel_compiler))]
-thread_local! {
-    /// A thread local variable that stores a pointer to the current `ImplicitCtxt`.
-    static TLV: Cell<usize> = const { Cell::new(0) };
-}
+    #[cfg(parallel_compiler)]
+    #[inline]
+    pub(super) fn set_tlv(value: usize) {
+        rayon_core::tlv::set(value);
+    }
 
-/// Sets TLV to `value` during the call to `f`.
-/// It is restored to its previous value after.
-/// This is used to set the pointer to the new `ImplicitCtxt`.
-#[cfg(not(parallel_compiler))]
-#[inline]
-fn set_tlv<F: FnOnce() -> R, R>(value: usize, f: F) -> R {
-    let old = get_tlv();
-    let _reset = rustc_data_structures::OnDrop(move || TLV.with(|tlv| tlv.set(old)));
-    TLV.with(|tlv| tlv.set(value));
-    f()
-}
+    /// Sets Rayon's thread-local variable, which is preserved for Rayon jobs
+    /// to `value` during the call to `f`. It is restored to its previous value after.
+    /// This is used to set the pointer to the new `ImplicitCtxt`.
+    #[cfg(parallel_compiler)]
+    #[inline]
+    pub(super) fn with_tlv<F: FnOnce() -> R, R>(value: usize, f: F) -> R {
+        rayon_core::tlv::with(value, f)
+    }
 
-/// Gets the pointer to the current `ImplicitCtxt`.
-#[cfg(not(parallel_compiler))]
-#[inline]
-fn set_tlv_raw(value: usize) {
-    TLV.with(|tlv| tlv.set(value));
-}
+    #[cfg(not(parallel_compiler))]
```

Comment:
if they're getting grouped like this would it make sense to instead do

```rust
#[cfg(parallel_compiler)]
mod tlv{ ... }
#[cfg(not(parallel_compiler)]
mod tlv { ... } 
```
or is there a good reason for marking all the individual items with `#[cfg()]` ?

Reply:
that would probably make more sense yeah